packages/records/lib/models/records.ts

```diff
@@ -433,3 +428,11 @@ async function getRecordsByExternalIds(external_ids: string[], connection_id: nu
 
     return result;
 }
+
+function shouldRetryUpsert(error: Error): boolean {
+    if ('code' in error) {
+        const errorCode = (error as { code: string }).code;
+        return errorCode === '40P01'; // deadlock_detected
```

Comment:
nit: seems a bit random to have this here

Reply:
I had it inlined in `upsert` originally. I am happy to put it back there if you prefer

Reply:
If there will be other error types that we should retry on then I would leave it here. If this is a one off then would inline it. Either way is fine to me tbh

Reply:
back to being inlined. thx